### PR TITLE
Add configure support to enable RAM class persistence

### PR DESCRIPTION
**Pull request by ThanHenderson**
Related: https://github.ibm.com/runtimes/openj9/issues/755
Related: https://github.com/eclipse-openj9/openj9/pull/20387
Signed-off-by: Nathan Henderson <nathan.henderson@ibm.com>

### CODE REVIEW

File: closed/autoconf/custom-spec.gmk.in [hunk 1]
```diff
@@ -75,6 +75,7 @@ OPENJ9_ENABLE_CRIU_SUPPORT           := @OPENJ9_ENABLE_CRIU_SUPPORT@
 OPENJ9_ENABLE_DDR                    := @OPENJ9_ENABLE_DDR@
 OPENJ9_ENABLE_DEMOS                  := @OPENJ9_ENABLE_DEMOS@
 OPENJ9_ENABLE_INLINE_TYPES           := @OPENJ9_ENABLE_INLINE_TYPES@
+OPENJ9_ENABLE_SNAPSHOTS              := @OPENJ9_ENABLE_SNAPSHOTS@
```
Comment (keithc-ca) [inline closed/autoconf/custom-spec.gmk.in:78]: Please move this after line 80.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: closed/autoconf/custom-hook.m4 [hunk 2]
```diff
@@ -47,6 +47,7 @@ AC_DEFUN_ONCE([CUSTOM_EARLY_HOOK],
   OPENJ9_CONFIGURE_DEMOS
   OPENJ9_CONFIGURE_HEALTHCENTER
   OPENJ9_CONFIGURE_INLINE_TYPES
+  OPENJ9_CONFIGURE_SNAPSHOTS
```
[the remark this answers is not in the document]
Comment (ThanHenderson) [reply closed/autoconf/custom-hook.m4:50]: I hadn't noticed the alphabetical order; I added it there following the location where the most recently added option (JFR) was added to. But I see the ordering now, and will adjust all the PRs accordingly.

[the remark this answers is not in the document]
Comment (ThanHenderson) [reply closed/autoconf/custom-hook.m4:50]: I'll reorder what you've mentioned here in the order you specified. For the `AC_DEFUN` blocks, I will place them in order relative to the option immediately prior. e.g. in this file I will put the `SNAPSHOT` block under the `NUMA` block despite those blocks not being in order.

In OpenJ9.gmk, there is no `NUMA` block, so I will respect the ordering specified in `custom-spec.gmk.in` and place the `SNAPSHOT` specifications beneath the `JITSERVER` blocks.

[the remark this answers is not in the document]
Comment (ThanHenderson) [reply closed/autoconf/custom-hook.m4:50]: Any other reordering should be done separately.

**Comment by ThanHenderson** [comment]
Let me know if you have any other comments. If not, I will apply these same changes to the other extension repo patches.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**